src/term/mod.rs

```diff
@@ -455,14 +455,20 @@ impl<'a> Iterator for RenderableCellsIter<'a> {
 
             // If selection color wasn't set by user we should invert cell colors.
             let selection_background_color = self.config.selection_background_color();
+            let selection_text_color = self.config.selection_text_color();
             let invert_selection = selected && selection_background_color.is_none();
 
             // Set custom selection color and selection text color, if they were provided.
             let (bg_color, fg_color) = if selected {
-                (
-                    selection_background_color.unwrap_or_else(|| cell.bg ),
-                    self.config.selection_text_color().unwrap_or_else(|| cell.fg),
-                )
+                // If selection text color was set by the user we shouldn't invert it.
+                if invert_selection && selection_text_color.is_some() {
+                    (selection_text_color.unwrap(), cell.fg)
```

Comment:
We should use the `let` pattern here to skip the unwrap.

Reply:
Are you talking about ```if let```? In this case it was ignored because rust does not allow writing
```rust
if let Some(x) = x && true
```

Reply:
Nested `if`s would still be better than unwraps which might lead to bugs down the road.

If nesting is not desired, there's also a workaround that can be applied:

```
if let ((true, Some(value))) = (invert_selection, selection_text_color.is_some())
```

Reply:
What do you think about this one?
```rust
let (bg_color, fg_color) = if selected {
    match (selection_background_color, self.config.selection_text_color()) {
        (None, None) => (cell.bg, cell.fg),
        (None, Some(fg_color)) => (fg_color, cell.fg),
        (Some(bg_color), None) => (bg_color, cell.fg),
        (Some(bg_color), Some(fg_color)) => (bg_color, fg_color),
    }
} else {
    (cell.bg, cell.fg)
};
```

Reply:
There's a lot of duplication in this one. 4/5 times the `fg_color` is just `cell.fg` for example, so that doesn't look like a great choice.

I think your current solution might actually make things harder on you than it has to be, I've played around with it myself for a bit and the following code worked great:

```rust
// Apply inversion and lookup RGB values
let mut fg_rgb = self.compute_fg_rgb(cell.fg, &cell);
let mut bg_rgb = self.compute_bg_rgb(cell.bg);

let selection_background = self.config.colors().selection.background;
let bg_alpha = if let (true, Some(col)) = (selected, selection_background) {
    bg_rgb = col;
    1.0
} else if selected ^ cell.inverse() {
    mem::swap(&mut fg_rgb, &mut bg_rgb);
    self.compute_bg_alpha(cell.fg)
} else {
    self.compute_bg_alpha(cell.bg)
};

if let (true, Some(col)) = (selected, self.config.colors().selection.text) {
    fg_rgb = col;
}
```

This would also make it possible to just remove all code from `color.rs` and remove the methods from the `config/mod.rs`. I think it would significantly clean things up here.

Reply:
I like your idea, since it's much easier to maintain. In my PR I just wanted to be consistent with the existing code for cursor colors specification ( e.g. method `selection_selection_color` in the first version of this PR  was named after [`cursor_cursor_color`](https://github.com/jwilm/alacritty/blob/e240da9ab3b819a8845ced1ab72d0a4239eac789/src/config/mod.rs#L1855)). But if you consider such a difference between implementations in almost identical things, then I would prefer your option.

Reply:
I don't think the named colors have a slightly different background and are mainly focused on the traditional 256 (/257/88) terminal colorspace.

Going for the simple solution here should be fine.

Reply:
Ok. So, waiting for tests and your review.